examples/bh_threaded.c

```diff
@@ -50,29 +55,49 @@ static int __init bottomhalf_init(void)
 
     pr_info("%s\n", __func__);
 
-    /* register LED gpios */
+/* register LED gpios */
+#ifdef GPIO_NO_REQUEST_ARRAY
+    ret = gpio_request(leds[0].gpio, leds[0].label);
+#else
     ret = gpio_request_array(leds, ARRAY_SIZE(leds));
+#endif
 
     if (ret) {
         pr_err("Unable to request GPIOs for LEDs: %d\n", ret);
         return ret;
     }
 
-    /* register BUTTON gpios */
+/* register BUTTON gpios */
+#ifdef GPIO_NO_REQUEST_ARRAY
+    ret = gpio_request(buttons[0].gpio, buttons[0].label);
+
+    if (ret) {
+        pr_err("Unable to request GPIOs for BUTTONs: %d\n", ret);
+        goto Button0Error;
+    }
+
+    ret = gpio_request(buttons[1].gpio, buttons[1].label);
+
+    if (ret) {
+        pr_err("Unable to request GPIOs for BUTTONs: %d\n", ret);
+        goto Botton1Error;
+    }
+#else
     ret = gpio_request_array(buttons, ARRAY_SIZE(buttons));
 
     if (ret) {
         pr_err("Unable to request GPIOs for BUTTONs: %d\n", ret);
-        goto fail1;
+        goto BottonError;
     }
+#endif
 
     pr_info("Current button1 value: %d\n", gpio_get_value(buttons[0].gpio));
 
     ret = gpio_to_irq(buttons[0].gpio);
 
     if (ret < 0) {
         pr_err("Unable to request IRQ: %d\n", ret);
-        goto fail2;
+        goto IRQ0Error;
```

Comment:
`IRQ0Error` was not better.

Reply:
I originally thought fail1, fail2, and fail3 would reduce the program's readability, but it has been fixed back.